packages/apps/human-app/frontend/src/pages/worker/jobs/components/escrow-address-search-form.tsx

```diff
@@ -0,0 +1,89 @@
+import Search from '@mui/icons-material/Search';
+import InputAdornment from '@mui/material/InputAdornment';
+import { FormProvider, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { useEffect } from 'react';
+import Grid from '@mui/material/Grid';
+import debounce from 'lodash/debounce';
+import { useColorMode } from '@/hooks/use-color-mode';
+import { Input } from '@/components/data-entry/input';
+import { addressSchemaOrEmptyString } from '@/shared/helpers/validate-address-schema';
+import { useIsMobile } from '@/hooks/use-is-mobile';
+
+interface SearchFormProps {
+  label: string;
+  placeholder: string;
+  columnId: string;
+  fullWidth?: boolean;
+  updater?: (fieldValue: string) => void;
+}
+
+export function EscrowAddressSearchForm({
+  label,
+  placeholder,
+  updater,
+  fullWidth = false,
+}: SearchFormProps) {
+  const isMobile = useIsMobile();
+  const { colorPalette } = useColorMode();
+  const methods = useForm<{ searchValue: string }>({
+    defaultValues: {
+      searchValue: '',
+    },
+    resolver: zodResolver(
+      z.object({ searchValue: addressSchemaOrEmptyString })
+    ),
+  });
+
+  const debouncedUpdater = debounce((value: string) => {
+    if (updater) updater(value);
```

Comment:
Maybe instead of doing this check we make `updated` required parameter for this, wdyt?

Reply:
done